src/shared/telemetry/defaultTelemetryService.ts

```diff
@@ -59,6 +67,7 @@ export class DefaultTelemetryService implements TelemetryService {
 
     public async start(): Promise<void> {
         this._eventQueue.push(...DefaultTelemetryService.readEventsFromCache(this.persistFilePath))
+        this._endOfCache = this._eventQueue[this._eventQueue.length - 1]
```

Comment:
Is there any chance that a metric can be added to the queue between the toolkit reload and here?

Reply:
Yep, and those must be allowlisted in `activeAllowed`: https://github.com/aws/aws-toolkit-vscode/pull/1613/files#diff-b7208a0816c68d584ca5740f07abef4aefc4fbbfaa12f4be5cf6916a407034adR281